src/Auth.ts

```diff
@@ -410,7 +499,18 @@ export class Auth {
       deviceCodeCallback: response => this.processDeviceCodeCallback(response, logger, debug),
       scopes: [`${resource}/.default`]
     };
-    return (this.clientApplication as Msal.PublicClientApplication).acquireTokenByDeviceCode(this.deviceCodeRequest) as Promise<AccessToken | null>;
+
+    const result = await (this.clientApplication as Msal.PublicClientApplication).acquireTokenByDeviceCode(this.deviceCodeRequest);
+
+    if (result === null) {
+      return null;
+    }
+
+    return {
+      accessToken: result.accessToken,
+      expiresOn: result.expiresOn,
+      identityId: result.account!.localAccountId
```

Comment:
I'm not sure, that's the difficulty with these msal typings. It makes you wonder sometimes if they are correct, there's no situation that I know of where the account is empty. Or maybe I've not looked hard enough. I'll recheck

Reply:
Could the account be empty for app-only auth?

Reply:
Ah, that's correct, for app only they are always null. Which could be the reason they use a nullable property, as the same AuthenticationResult interface is used.

In that case it would be ok to use a `!` here. 

An alternative would be to always extract the `oid` from the token like I've done with app identities.